srcCxx/shape_main.cxx

```diff
@@ -314,254 +332,234 @@ class ShapeOptions {
-                                        + std::to_string(shapesize),
+                if (optarg[0] != '\0') {
+                    switch (optarg[0]) {
+                    case '1':
+                        data_representation = XCDR_DATA_REPRESENTATION;
+                        break;
+                    case '2':
+                        data_representation = XCDR2_DATA_REPRESENTATION;
+                        break;
+                    default:
+                        logger.log_message("unrecognized value for data representation "
+                                    + std::string(1, optarg[0]),
                                 Verbosity::ERROR);
                         parse_ok = false;
                     }
-                    break;
                 }
-            case '?':
-                {
+                break;
+            case 'z': {
+                int converted_param = sscanf(optarg, "%d", &shapesize);
+                if (converted_param == 0) {
+                    logger.log_message("unrecognized value for shapesize "
+                                + std::string(1, optarg[0]),
+                            Verbosity::ERROR);
+                    parse_ok = false;
+                } else if (shapesize < 0) {
+                    logger.log_message("incorrect value for shapesize "
+                                    + std::to_string(shapesize),
+                            Verbosity::ERROR);
                     parse_ok = false;
-                    break;
                 }
+                break;
+            }
+            case 'W': {
+                int converted_param = 0;
+                sscanf(optarg, "%d", &converted_param);
+                if (sscanf(optarg, "%d", &converted_param) == 0) {
+                    logger.log_message("unrecognized value for write-period "
+                                + std::string(1, optarg[0]),
+                            Verbosity::ERROR);
+                    parse_ok = false;
+                }
+                write_period_us = (useconds_t) converted_param * 1000;
+                if (write_period_us < 0) {
+                    logger.log_message("incorrect value for write-period "
+                                + std::to_string(write_period_us),
+                            Verbosity::ERROR);
+                    parse_ok = false;
+                }
+                break;
+            }
+            case 'A': {
+                int converted_param = 0;
+                sscanf(optarg, "%d", &converted_param);
+                if (sscanf(optarg, "%d", &converted_param) == 0) {
+                    logger.log_message("unrecognized value for read-period "
+                                + std::string(1, optarg[0]),
+                            Verbosity::ERROR);
+                    parse_ok = false;
+                }
+                read_period_us = (useconds_t) converted_param * 1000;
+                if (read_period_us < 0) {
```

Comment:
As @mitza-oci pointed out, this will give a warning since `read_period_us` is unsigned